src/redux/reducers/bookmarkReducers.js

```diff
@@ -0,0 +1,36 @@
+import {
+  BOOKMARK_ARTICLE,
+  REQUEST_SUBMITTED,
+  REQUEST_FAILED
+} from "../actionTypes";
+
+export const initialState = {
+  isSubmitting: false,
+  message: "",
+  bookmarkedArticles: {}
+};
+
+export default (state = initialState, action) => {
+  switch (action.type) {
+    case BOOKMARK_ARTICLE:
+      return Object.assign(state, {
+        bookmarkedArticles: Object.assign(state.bookmarkedArticles, {
+          [action.payload.articleSlug]: action.payload.isBookmarked
+        }),
+        isSubmitting: false
+      });
```

Comment:
I believe that using this format `  return {
        ...state,
        bookmarkedArticles: {
          [action.payload.articleSlug]: action.payload.isBookmarked
        },
        isSubmitting: false
      };` can result to more clean syntax than using `Object.assign()` method.

Reply:
Okay, I have removed it ...
Thanks.

Reply:
Fixed